rfcs/0006-gtfs-canonical-routes.md

```diff
@@ -0,0 +1,108 @@
+- Asana task: [asana link](https://app.asana.com/)
+- Status: Proposed
+
+# Summary
+
+This will enable us to publish in our static GTFS feed "canonical" data for our rail routes. We will
+do so via a "canonical" service that's never active, but which has representative trips, stop times,
+shapes, and route patterns.
+
+# Motivation
+
+MBTA.com users are able to see line diagrams that match real-world state, including physical maps,
+even when stations are closed for a long duration.
+
+PIOs are able to create alerts for stations without any scheduled service, and T-Alerts subscribers
+are able to stay informed about the status of the station closure.
+
+# Guide-level explanation
+
+GTFS output includes additional, hardcoded data about where rail lines go to and where they can stop
+if all stations were open, which is independent of all scheduled service.
+
+This is represented as a service with ID "canonical", which is valid for the entire feed date range,
+but which has no active days of the week. The typicality of the service is tagged with a new value
+so it can be programmatically identified.
+
+All rail routes have two or more trips on this service, corresponding to the "canonical" patterns of
+service in each direction for any branches, as understood by stakeholders. Each such trip has a
+route pattern, for which it is the representative trip, and a shape, and the route pattern is
+considered "typical" for the route. If current service has trips following the same route pattern,
+they will use the canonical route pattern as well.
+
+In short, for consumers of the feed, a route pattern has typicality of `1` if and only if it visits
+a set of canonical stops in a canonical order.
+
+# Reference-level explanation
+
+The canonical service appears in the following GTFS files in the following way:
+
+- **calendar.txt** includes a new `service_id` to envelop all of the canonical trips on all the
+  routes. `start_date` and `end_date` should reflect the feed's valid start/end dates. All other
+  fields should have a value of `0`.
+- **calendar_attributes.txt** includes the new `service_id`. `service_schedule_type` should be
+  `"Other"`, and `rating_start_date` and `rating_end_date` should reflect the feed's valid start/end
+  dates. `service_schedule_typicality` should be `6`, and the
+  [docs](https://github.com/mbta/gtfs-documentation/blob/master/reference/gtfs.md) should be updated
+  with this new kind of value.
+- **route_patterns.txt** includes new route patterns for all routes/directions we want to hardcode.
+  `route_pattern_typicality` should be `1`, and `route_pattern_sort_order` should be a less than the
+  value for all other route patterns for that given route/direction. For routes with branches,
+  you'll need a pattern for each route/direction/branch (eg, for Providence/Stoughton Line, you
+  should have two new route patterns in the outbound direction).
+- **shapes.txt** can probably use the existing shapes.
```

Comment:
This won't always work for Subway, since the shapes (and `shape_id`s) can change between ratings. We might want to create our own duplicate shapes and store them in `input/canonical/shapes.txt`.